components/locid_transform/src/provider.rs

```diff
@@ -135,6 +135,56 @@ pub struct AliasesV1<'data> {
     pub subdivision: ZeroMap<'data, UnvalidatedSubdivision, SemivalidatedSubdivision>,
 }
 
+// TODO: move to directionality module when stabilizing
+/// Represents the direction of a script.
+///
+/// [`LocaleDirectionality`] can be used to get this information.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[zerovec::make_ule(DirectionULE)]
+#[zerovec::derive(Debug)]
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[cfg_attr(
+feature = "datagen",
+derive(serde::Serialize, databake::Bake),
+databake(path = icu_locid_transform::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[non_exhaustive]
+pub enum Direction {
+    /// The script is left-to-right.
+    LeftToRight = 0,
+    /// The script is right-to-left.
+    RightToLeft = 1,
+}
+
+#[icu_provider::data_struct(ScriptDirectionV1Marker = "locid_transform/script_dir@1")]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_locid_transform::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+/// This directionality data is used to determine the script directionality of a locale.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[yoke(prove_covariance_manually)]
+pub struct ScriptDirectionV1<'data> {
+    /// Script directionality.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub rtl: ZeroMap<'data, UnvalidatedScript, Direction>,
```

Comment:
thought: would be cool to be able to bitpack the direction array. not worth trying here though